### PR TITLE
Search Block: Apply font-related style inheritance to input field

### LINKED ISSUE #60174: Custom fonts aren't applied to the input placeholder of the Search Block
## Description

With the Font Library feature enabled, users can apply custom fonts to their websites. However, the custom font is not being correctly applied to the placeholder text within the Search Block.

## Expected behaviour

The Search Block should use the selected font.

## WordPress information
- WordPress version: WordPress 6.5 - RC3

image

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #60174

## What?

This PR adds CSS properties to the input element inside the Search block that inherits font-related styles. This ensures that when typography styles are defined at **the root level**, they are correctly applied to the input element as well.

The screenshot below is from Emptytheme:

### Before

image

### After

image

## Why?

According to [MDN](https://developer.mozilla.org/en-US/docs/Learn/Forms/Styling_web_forms#fonts_and_text), form-related elements are browser-specific and do not inherit font-related styles. This is not an issue when typography-related styles are applied at the block level in the global styles or on individual blocks, but it becomes an issue when typography-related settings are applied at the root level in the global styles.

## How?

Applies CSS to the input element that inherits all font-related styles. We also use :where to reduce the CSS specificity so that we can override this style. Only `text-decoration` is not inherited. This is because it causes problems with text input, as discussed in  #43499.

Ideally, it would be more convenient to define these styles for the input field in the default theme.json, but the current theme.json does not support input elements.

## Testing Instructions

- Insert a Search block into one of templates.
- Apply typography related styles at the root level via the global styles. Those styles should now be applied correctly to the input element, including the placeholder.
- Apply typography-related styles to the Search block via the global styles. All styles should be overridden correctly.
- Apply typography-related styles to the individual Search block. All styles should be overridden correctly.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/66afc426-52cf-46a9-96f4-026fbb3eef71

### CODE REVIEW

File: packages/block-library/src/search/style.scss [hunk 1]
```diff
@@ -97,6 +97,15 @@ $button-spacing-y: math.div($grid-unit-15, 2); // 6px
 	}
 }
 
+// We are lowering the specificity so that the input element can override the rule for the input element inside the search block.
+:where(.wp-block-search__input) {
+	font-family: inherit;
+	font-weight: inherit;
+	font-size: inherit;
+	line-height: inherit;
+	letter-spacing: inherit;
```
Comment (ramonjd) [inline packages/block-library/src/search/style.scss:106]:
```suggestion
	letter-spacing: inherit;
	text-transform: inherit;
```

**Review by ramonjd — APPROVED** [review]
Thanks for this PR @t-hamano 

| Before | After |
|--------|--------|
| Screenshot 2024-04-02 at 9 13 20 am | Screenshot 2024-04-02 at 9 14 16 am | 





Things are working as described. I think we're missing one declaration `text-transform: inherit;`

`text-decoration` is unset here:

https://github.com/WordPress/gutenberg/blob/d7b3963a0691ebe8fb434289b2bea98eb6cc6883/packages/block-library/src/search/style.scss#L49

So I guess we don't need that.

File: packages/block-library/src/search/style.scss (hunk 1 above)
Comment (aaronrobertshaw) [reply packages/block-library/src/search/style.scss:106]:
```suggestion
	letter-spacing: inherit;
	text-transform: inherit;
	font-style: inherit;
```

I think we're missing one more typography related style. Text decoration and text columns are rightfully omitted.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for this enhancement @t-hamano 👍 

It's testing mostly as advertised. There's still a couple of typography related styles missing though.

- Text transform (as Ramon noted)
- Font style

You can see the italic font style getting picked up by the search block's label but not the input in the screenshot below:

Screenshot 2024-04-02 at 1 46 39 PM

**Comment by t-hamano** [comment]
Thank you for noticing what I had missed! I added the missing styles and updated the screenshot as well.

**Review by aaronrobertshaw — APPROVED** [review]
Appreciate the quick iteration @t-hamano 👍 

The input is inheriting all the applicable typography styles now on both the frontend and in the editor.

I did notice one inconsistency between the editor and frontend for the button though. The search block's button doesn't inherit the text transform style on the frontend. Taking a closer look at the button's styles might be a good follow-up.

| Editor | Frontend |
|---|---|
| Screenshot 2024-04-02 at 3 27 22 PM | Screenshot 2024-04-02 at 3 27 29 PM |

**Comment by ramonjd** [comment]
Nice catch on `font-style` too @aaronrobertshaw

**Comment by t-hamano** [comment]
@aaronrobertshaw @ramonjd Thanks for the review!

> I did notice one inconsistency between the editor and frontend for the button though. The search block's button doesn't inherit the text transform style on the frontend. Taking a closer look at the button's styles might be a good follow-up.

Thank you for finding it. I would like to file an issue regarding this point.

**Pull request merged**